src/pkynetics/technique_analysis/dilatometry.py

```
@@ -312,23 +539,58 @@
+    if len(valid_fraction) == 0:
+        # Fallback: linear interpolation between start and end points
+        from scipy.interpolate import interp1d
+
+        try:
+            temps = np.array([start_temp, end_temp])
+            fracs = np.array([0.0, 1.0])
+            if is_cooling:
+                fracs = np.array([1.0, 0.0])
+            f = interp1d(fracs, temps)
+            return float(f(0.5))
+        except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)